src/Core/Batch/SysvConfigStorage.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\Batch;
+
+/**
+ * ConfigStorageInterface implementation with SystemV IPC shared memory.
+ */
+class SysvConfigStorage implements ConfigStorageInterface
+{
+    const VAR_KEY = 1;
+
+    /* @var int */
+    private $sysvKey;
+
+    /* @var int */
+    private $semid;
+
+    /**
+     * Prepare the key for semaphore and shared memory.
+     */
+    public function __construct()
+    {
+        $this->sysvKey = ftok(__FILE__, 'A');
+        $this->semid = sem_get($this->sysvKey, 1, 0600, 1);
+    }
+
+    /**
+     * Acquire a lock.
+     *
+     * @return boolean
+     */
+    public function lock()
+    {
+        return sem_acquire($this->semid);
```

Comment:
In the spanner PR we are introducing the [symfony lock component](https://github.com/symfony/lock). Would it be worthwhile for us to extract that out and introduce it sooner so it can be leveraged here as well?